dspy/streaming/streaming_listener.py

```diff
@@ -208,13 +208,42 @@ def flush(self) -> str:
             return last_tokens[:boundary_index]
         elif isinstance(settings.adapter, ChatAdapter) or settings.adapter is None:
             boundary_index = last_tokens.find("[[")
+            if boundary_index == -1:
+                # No end marker found, return all tokens
+                return last_tokens
```

Comment:
let's be consistent with other condition block: `boundary_index = len(last_tokens)`